crates/ruff_diagnostics/src/fix.rs

```diff
@@ -0,0 +1,43 @@
+use rustpython_parser::ast::Location;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::edit::Edit;
+
+/// A collection of edits to be applied to a source file.
+#[derive(Default, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Fix {
+    edits: Vec<Edit>,
+}
+
+impl Fix {
+    pub fn new(edits: Vec<Edit>) -> Self {
+        // TODO(charlie): Ensure that the edits are sorted and non-overlapping.
+        debug_assert!(!edits.is_empty(), "Fix must have at least one edit");
+
+        Self { edits }
+    }
+
+    pub fn none() -> Self {
+        Self::default()
+    }
+
+    pub fn is_none(&self) -> bool {
+        self.edits.is_empty()
+    }
+
+    pub fn location(&self) -> Location {
+        self.edits[0].location
+    }
```

Comment:
This will panic on `Fix::default` and `Fix::none()`. It also isn't clear for a caller what to expect as the result of `fix.location()`. That's why I would change the method to

```suggestion
    pub fn first_location(&self) -> Option<Location> {
        self.edits.first().copied()
    }
```

Reply:
looking at crates/ruff/src/autofix/mod.rs, shouldn't location be a new location (min of all starts, max of all ends), or maybe even something defined by the rule that selects the whole expression that is fixed so that we don't combine potentially incorrect fixes on the same expression?